### PR TITLE
adding FindAllProject and FindByIDproject

### PR DESCRIPTION
**Pull request by jhonlyoyd**

### CODE REVIEW

File: src/main/java/com/csv/communitytrackerjava/service/ProjectServiceImpl.java [hunk 1]
```diff
@@ -76,14 +79,19 @@ public ProjectResponseDTO updateProject(ProjectUpdateDTO projectUpdateDTO, Integ
...
         return toProjectResponseDTO("Successfully update project.", payloadDTO);
     }
 
     @Override
-    public ProjectResponseDTO findAllProject() {
+    public ProjectResponseDTO findAllProject(Pageable pageable) {
         ProjectPayloadDTO payloadDTO = new ProjectPayloadDTO();
-        payloadDTO.setAdditionalProperty("projects", projectRepository.findAll());
+        List<ProjectDTO> projectList = (
+                projectRepository.findByIsActiveTrue(pageable)
```
Comment (eldrichm-softvision) [inline src/main/java/com/csv/communitytrackerjava/service/ProjectServiceImpl.java:94]: Do we really want to get only active project here?

File: src/main/java/com/csv/communitytrackerjava/controller/ProjectController.java [hunk 2]
```diff
@@ -25,8 +25,8 @@ public class ProjectController {
 
 
     @GetMapping
-    public ResponseEntity<ProjectResponseDTO> findAll() {
-        return new ResponseEntity<>(projectService.findAllProject(), HttpStatus.OK);
+    public ResponseEntity<ProjectResponseDTO> findAll(Pageable pageable) {
```
Comment (eldrichm-softvision) [inline src/main/java/com/csv/communitytrackerjava/controller/ProjectController.java]: add test method to handle pageable request.

File: src/main/java/com/csv/communitytrackerjava/repository/ProjectRepository.java [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 import com.csv.communitytrackerjava.model.Project;
 import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
```
Comment (eldrichm-softvision) [inline src/main/java/com/csv/communitytrackerjava/repository/ProjectRepository.java]: Query not used remove

File: src/main/java/com/csv/communitytrackerjava/service/ProjectServiceImpl.java [hunk 4]
```diff
@@ -76,14 +79,19 @@ public ProjectResponseDTO updateProject(ProjectUpdateDTO projectUpdateDTO, Integ
...
         modelMapper.map(projectUpdateDTO, projectFound);
-        payloadDTO.setAdditionalProperty("projects", projectMapper.toDTO(projectRepository.save(projectFound)));
+        payloadDTO.setAdditionalProperty("projects", projectMapper
+                .toDTO(projectRepository.save(projectFound)));
         return toProjectResponseDTO("Successfully update project.", payloadDTO);
     }
 
     @Override
-    public ProjectResponseDTO findAllProject() {
+    public ProjectResponseDTO findAllProject(Pageable pageable) {
```
Comment (eldrichm-softvision) [inline src/main/java/com/csv/communitytrackerjava/service/ProjectServiceImpl.java]: Add test method for the changes

File: src/main/java/com/csv/communitytrackerjava/service/ProjectServiceImpl.java (hunk 1 above)
Comment (jhonlyoyd) [reply src/main/java/com/csv/communitytrackerjava/service/ProjectServiceImpl.java:94]: yes for the testing of the delete.

File: src/main/java/com/csv/communitytrackerjava/repository/ProjectRepository.java (hunk 3 above)
Comment (eldrichm-softvision) [reply src/main/java/com/csv/communitytrackerjava/repository/ProjectRepository.java]: please remove this

File: src/test/java/com/csv/communitytrackerjava/service/ProjectServiceGetProjectTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    @SneakyThrows
+    @Test
+    public void testFindProjectById() {
+        Mockito.when(projectRepository.findById(anyInt()))
+                .thenReturn(Optional.ofNullable(sample1));
+
+        ProjectResponseDTO result = projectService.findProjectById(1);
+
+        ProjectResponseDTO expected = new ProjectResponseDTO();
```
Comment (eldrichm-softvision) [inline src/test/java/com/csv/communitytrackerjava/service/ProjectServiceGetProjectTest.java:87]: Add verify as you have mock here.

**Review by eldrichm-softvision — APPROVED** [review]

**Pull request merged**